lib/types.nix

```diff
@@ -113,4 +113,32 @@ rec {
         types.optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
       }";
     };
+
+  pluginLuaConfig = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        pre = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration before the initialization of the plugin";
+        };
+        post = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration after the initialization of the plugin";
+        };
+        init = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Initialization code (between pre & post)";
+        };
+      };
+
+      config.init = lib.mkMerge [
```

Comment:
Interesting, so the `mkMerge` combines `pre` and `post` with `init` instead of replacing its contents in this declaration?

Reply:
Yeah, I added a test to check that it indeed works. I was inspired with what @MattSturgeon did